README.md

```diff
@@ -90,9 +158,169 @@ reproducible.
+you can write pure Nix expressions in **garn**, and import Nix projects, but
+you can also from Nix call things generated by **garn**.
 
 ## Typescript API
 
-Documentation for the `garn` Deno library is documented
-[here](https://doc.deno.land/https://garn.io/ts/v0.0.18/mod.ts).
+You can find documentation for the **garn** Deno library [here](https://doc.deno.land/https://garn.io/ts/v0.0.18/mod.ts).
+
+
+## Comparison to other tools
+
+- **Nix**: Nix is a brilliant idea with an immense amount of volunteer effort
+  behind it, and the giant shoulders on which **garn** stands. It has a rather
+  complex CLI, and the Nix language is a whole new thing to learn. That said
+  it is a much more mature project with support for just about every imaginable
+  stack. For complex projects, you're almost always better off picking Nix.
+  For simpler ones, or ones that include collaborators who don't feel comfortable
+  with Nix, **garn** might be a better option.
+
+  But the choice isn't exclusive
+  or binary: **garn** it both allows including
+  Nix expressions or dependencies within it, and generates a `flake.nix` that
+  can be consumed by other Nix expressions, so that you can combine Nix and
+  **garn** quite well.
+- **just**, **taskfile**, and **make**: These are build tools or command runners
+  that help you write and organize the scripts or commands of your project.
+  In that regard, they overlap with **garn**.
+  They are much easier to install and get started with than **garn**. But they
+  don't manage packages and environments at all, so every user still needs to
+  figure that out themselves. **garn** also makes it easy to share your "recipes"
+  as a library. (make also does incremental builds, which **garn** does not.)
+- **virtualenv**, **rvm**, **nvm**, etc.: These are *language-specific* tools;
+  very helpful if you need a particular python package or version of python,
+  not at all if you need a system package.
+- **devbox** and **devenv**: Both of these projects are more mature than **garn**;
+  you are likely to find fewer rough corners with them.
+  Other than that, **garn** differs from them in three major ways: it supports
+  writing your own packages and checks which run in isolated environments (such as CI)
+  and can be cached or consumed by downstream users; it allows configuration in
+  Typescript rather than YAML and/or Nix; and it allows easily writing and importing
+  libraries with reusable functionality.
+- **docker**: Docker isolates processes, and is often used to distribute dev
+  environments. **garn** overlaps with this feature . Both tools make it relatively certain that if it works once
+  it works everywhere. But with Docker that comes at the price of harder
+  integration with your development process.
+
+  Besides development environments, Docker is also used for other purposes, such
+  as deployment. **garn** does not support any of that (yet).
```

Comment:
Can we maybe run word-wrapping on these text snippets? The line breaks are a bit erratic. (This doesn't matter for rendered markdown, but for ascii.)